x/mongo/driver/operation/count.go

```diff
@@ -54,15 +56,38 @@ func buildCountResult(response bsoncore.Document, srvr driver.Server) (CountResu
 	cr := CountResult{}
 	for _, element := range elements {
 		switch element.Key() {
-		case "n":
+		case "n": // for count using original command
 			var ok bool
 			cr.N, ok = element.Value().AsInt64OK()
 			if !ok {
-				err = fmt.Errorf("response field 'n' is type int64, but received BSON type %s", element.Value().Type)
+				err = fmt.Errorf("response field %s is type int64, but received BSON type %s",
+					element.Key(), element.Value().Type)
+				break
```

Comment:
nit: Remove any uses of `break` where it's the last statement in a `case` block because it doesn't have any effect.